libs/designer/src/lib/core/utils/parameters/helper.ts

```diff
@@ -1027,17 +1027,32 @@ export function getExpressionValueForOutputToken(token: OutputToken, nodeType: s
       }
 
     default:
-      method = arrayDetails ? constants.ITEM : getTokenExpressionMethodFromKey(key, actionName);
+      method = arrayDetails ? constants.ITEM : getTokenExpressionMethodFromKey(key, actionName, source);
       return generateExpressionFromKey(method, key, actionName, !!arrayDetails, !!required);
   }
 }
 
-export function getTokenExpressionMethodFromKey(key: string, actionName: string | undefined): string {
+export function getTokenExpressionMethodFromKey(key: string, actionName: string | undefined, source: string | undefined): string {
   const segments = parseEx(key);
   if (segmentsAreBodyReference(segments)) {
     return actionName ? `${OutputSource.Body}(${convertToStringLiteral(actionName)})` : constants.TRIGGER_BODY_OUTPUT;
   } else {
-    return actionName ? `${constants.OUTPUTS}(${convertToStringLiteral(actionName)})` : constants.TRIGGER_OUTPUTS_OUTPUT;
+    if (actionName) {
+      return `${OutputSource.Outputs}(${convertToStringLiteral(actionName)})`;
+    }
+    if (source) {
+      const propertyInQueries = !!source && equals(source, OutputSource.Queries);
+      const propertyInHeaders = !!source && equals(source, OutputSource.Headers);
+      const propertyInStatusCode = !!source && equals(source, OutputSource.StatusCode);
```

Comment:
Is there a reason to have the `!!source` on each line if we're checking it in the if statement to get into this block?

Reply:
whoops added the if block after, will fix this